### ISSUE TITLE
Top Padding is not respected by iOs AppBar on device

### ISSUE THREAD

**Issue by BenSower** [flutter/flutter#13536]
## Steps to Reproduce
I hope this is reproduceable:
1. Update to the latest alpha 0.0.20 (it looks like this was introduced in 0.0.18)
2. Run the gallery app on a iOs device (Not a simulator! For some reason the simulator seems to be working properly) 
3. Go to any screen including an app-bar, e.g. "buttons"
**Expected layout**:
![flutter_02](https://user-images.githubusercontent.com/1158039/33944603-fac7584a-e01c-11e7-8752-88a9e9c6fda6.png)

**Actual layout**:
![flutter_02](https://user-images.githubusercontent.com/1158039/33944676-251500e8-e01d-11e7-9cae-538d4cae8d25.png)

## Logs
If necessary I can also provide some logs, if this is not reproducable.

## Flutter Doctor
```
[✓] Flutter (on Mac OS X 10.11.6 15G17023, locale en, channel alpha)
    • Flutter at /Users/xy/flutter
    • Framework revision 8f65fec5f5 (21 hours ago), 2017-12-12 09:50:14 -0800
    • Engine revision edaecdc8b8
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d8ae797298c3a6cf8dc9f4558707bd2672224d3e

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 8.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.2.1, Build version 8C1002
    • ios-deploy 1.9.2
    • CocoaPods version 1.2.1

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] Connected devices
    • iPhone von xy 2 • 9c2ac8e5e8d9e412312357dddbef9042dd90 • ios • iOS 11.2
    • iPhone 5s          • 22754963-E2C1-49FA-B538-8F188E522910     • ios • iOS 10.2 (simulator)
```

**Comment by Hixie** [comment]
@xster @cbracken fallout from your recent work?

**Comment by cbracken** [comment]
@BenSower thanks for filing this! I'm able to repro this locally when building with Xcode 8. Under Xcode 9, things look correct on the devices I tested (iPhone X on iOS 11.2, iPod touch on iOS 10.3.3, iPhone SE on iOS 11.2).

Xcode 8 includes the iOS 10 SDK, which does not include support for the safe area inset APIs, which is what we use to compute window padding. Flutter is built with Xcode 9 against the iOS 11 SDK (with deployment target: iOS 8). Updating your Xcode install to Xcode 9 should resolve the issue for you.

I'll update our Xcode version check in flutter to make this more obvious in the flutter doctor output.

Thanks again for reporting this!

**Comment by BenSower** [comment]
Hi @cbracken thanks for your time and thank you for thoroughly reproducing the issue. If you had just said that everything is fine since you only tested Xcode 9 (which would have been totally fine) I'd basically gone crazy :-). Yeah, we are still using Xcode 8 in a very hacky way, since our old dev-mac is from 2008 and not able to run Xcode 9... 

Ok, we wanted to get a new machine anyways and now is as good a time as any.
Thank you for your time and for walking the extra mile, it is greatly appreciated!

P.S.: [this](https://flutter.io/faq/#what-devices-and-os-versions-does-flutter-run-on) (>iOs 8) still stands?

**Comment by cbracken** [comment]
> thanks for your time and thank you for thoroughly reproducing the issue.

@BenSower Happy to do it! As I say, the current flutter doctor check for Xcode 8+ makes this really unobvious. I'll get a patch out today to update the check.

> we are still using Xcode 8 in a very hacky way, since our old dev-mac is from 2008 and not able to run Xcode 9...

Not promising anything, but before you pick up a new Mac for development, if you're willing to use Xcode 8 in an even hackier way... give the following a try and you might be able to extend your machine's shelf life a little longer:

1. Download Xcode 9 from developer.apple.com > Downloads
2. Extract the .xip archive and move Xcode.app to /Applications/Xcode9.app
3. On the command-line, do the following:
```
cd /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs
ln -s /Applications/Xcode9.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk iPhoneOS11.2.sdk
```
4. Open your Flutter app workspace in Xcode. Select the project in the left sidebar.
5. Inside the settings view, there should be a left sidebar. Make sure the Runner *target* is selected.
6. Select the *Build Settings* tab.
7. Under the *Architectures* section, find *Base SDK*. Select iOS 11.2.
8. Try a build and hope it works!

I just tested this out locally and it worked for me with Flutter Gallery.

> P.S.: this (>iOs 8) still stands?

Yep! Xcode 9 with the iOS 11 SDK is what's needed for development, but we still target depolyment to iOS 8 and above. You can adjust that for your project in the *Deployment Target* field in your project settings in Xcode. Our templates default to iOS 8 as the deployment target.

**Comment by BenSower** [comment]
@cbracken yeah, I tried to do something like your version before (we were already using [this hack](https://stackoverflow.com/questions/44381860/use-xcode-8-with-ios-11) to be able to use xcode 8 to deploy on newer iOs devices, so I thought I'd try the same thing with the SDKs, too), but failed to even get XCode to properly recognize the newer SDK. Using your version I got it to recognize 11.2 as well as build it using Xcode, even though it seems like it didn't actually use 11.2, but defaulted back to 10, since the bug persisted. Weirdly enough I couldn't build the project at all using flutter directly :-/, which is weird since it uses xcode-build under the hood, too... 

Long story short: We found out, we still had a newer Mac flying around the office, so now I am sitting in a very nice cafe, enjoying the afternoon, setting up the new Mac :-). Things could be worse!

Thanks again for your work, at least I learned something about Xcode in the process. Take care!

**Comment by christian-muertz** [comment]
@cbracken hey, i tried your version of using the iOS 11 SDK in Xcode8, but i cant build the app using the flutter cli:

Iam using an iMac (Early 2009) running El Capitan with xCode8. Here are the logs:

### flutter build ios
```
Building de.teraarts.fluttercojobo for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: Y74TD5QW62
Running Xcode clean...
Starting Xcode build...
Xcode build done
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
    
Xcode's output:
↳
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIAccelerometer.h:8:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:128:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUUID.h:26:49: error: nullability specifier '_Nullable' cannot be applied to non-pointer type 'uuid_t' (aka 'unsigned char [16]')
    - (instancetype)initWithUUIDBytes:(const uuid_t _Nullable)bytes;
                                                    ^
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUUID.h:29:30: error: nullability specifier '_Nonnull' cannot be applied to non-pointer type 'uuid_t' (aka 'unsigned char [16]')
    - (void)getUUIDBytes:(uuid_t _Nonnull)uuid;
                                 ^
    2 errors generated.
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:11:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIAccelerometer.h:8:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'QuartzCore' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:9:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/QuartzCore.h:9:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CoreAnimation.h:10:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CATransform3D.h:12:9: fatal error: could not build module 'Foundation'
    #import <Foundation/NSValue.h>
     ~~~~~~~^
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'QuartzCore' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:9:
    While building module 'OpenGLES' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAEAGLLayer.h:7:
    In file included from <module-includes>:1:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/OpenGLES.framework/Headers/EAGL.h:8:10: fatal error: could not build module 'Foundation'
    #include <Foundation/Foundation.h>
     ~~~~~~~~^
    1 error generated.
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'QuartzCore' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:9:
    While building module 'Metal' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAMetalLayer.h:7:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Metal.framework/Headers/Metal.h:9:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLTypes.h:8:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'QuartzCore' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:9:
    While building module 'Metal' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAMetalLayer.h:7:
    While building module 'IOSurface' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLTexture.h:15:
    In file included from <module-includes>:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/IOSurface.framework/Headers/IOSurfaceObjC.h:15:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    1 error generated.
    2 errors generated.
    4 errors generated.
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'CoreImage' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIColor.h:11:
    In file included from <module-includes>:1:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreImage.framework/Headers/CoreImage.h:9:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'CoreImage' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIColor.h:11:
    While building module 'CoreVideo' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreVideo.framework/Headers/CoreVideo.h:29:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreVideo.framework/Headers/CVPixelBuffer.h:461:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreVideo.framework/Headers/CVPixelBufferIOSurface.h:26:10: fatal error: could not build module 'IOSurface'
    #include <IOSurface/IOSurfaceRef.h>
     ~~~~~~~~^
    1 error generated.
    2 errors generated.
    While building module 'UIKit' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:
    While building module 'FileProvider' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/FileProvider.framework/Headers/FileProvider.h:8:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/FileProvider.framework/Headers/NSFileProviderDomain.h:8:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    1 error generated.
    10 errors generated.
    /Users/christian/flutter-cojoboapp/ios/Runner/main.m:1:9: fatal error: could not build module 'UIKit'
    #import <UIKit/UIKit.h>
     ~~~~~~~^
    While building module 'Flutter' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:2:
    In file included from <module-includes>:1:
    In file included from /Users/christian/flutter-cojoboapp/ios/Flutter/Flutter.framework/Headers/Flutter.h:34:
    /Users/christian/flutter-cojoboapp/ios/Flutter/Flutter.framework/Headers/FlutterAppDelegate.h:8:9: fatal error: could not build module 'UIKit'
    #import <UIKit/UIKit.h>
     ~~~~~~~^
    While building module 'Flutter' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:2:
    While building module 'CoreMedia' imported from /Users/christian/flutter-cojoboapp/ios/Flutter/Flutter.framework/Headers/FlutterTexture.h:8:
    While building module 'AudioToolbox' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMedia.framework/Headers/CMFormatDescription.h:27:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AudioToolbox.h:18:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AUGraph.h:34:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AudioUnit.h:24:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AUAudioUnit.h:11:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AUParameters.h:13:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    While building module 'Flutter' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:2:
    While building module 'CoreMedia' imported from /Users/christian/flutter-cojoboapp/ios/Flutter/Flutter.framework/Headers/FlutterTexture.h:8:
    While building module 'AudioToolbox' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMedia.framework/Headers/CMFormatDescription.h:27:
    While building module 'CoreMIDI' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/MusicPlayer.h:44:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMIDI.framework/Headers/CoreMIDI.h:24:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMIDI.framework/Headers/MIDINetworkSession.h:20:9: fatal error: could not build module 'Foundation'
    #import <Foundation/Foundation.h>
     ~~~~~~~^
    1 error generated.
    2 errors generated.
    While building module 'Flutter' imported from /Users/christian/flutter-cojoboapp/ios/Runner/main.m:2:
    While building module 'CoreMedia' imported from /Users/christian/flutter-cojoboapp/ios/Flutter/Flutter.framework/Headers/FlutterTexture.h:8:
    In file included from <module-includes>:1:
    In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMedia.framework/Headers/CoreMedia.h:15:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreMedia.framework/Headers/CMFormatDescription.h:27:10: fatal error: could not build module 'AudioToolbox'
    #include <AudioToolbox/AudioFormat.h>
     ~~~~~~~~^
    3 errors generated.
    4 errors generated.
    15 errors generated.

```

### flutter doctor
```
[✓] Flutter (Channel beta, v0.1.5, on Mac OS X 10.11.6 15G1011, locale de)
    • Flutter version 0.1.5 at /Users/christian/flutter
    • Framework revision 3ea4d06340 (4 weeks ago), 2018-02-22 11:12:39 -0800
    • Engine revision ead227f118
    • Dart version 2.0.0-dev.28.0.flutter-0b4f01f759

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 8.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.2.1, Build version 8C1002
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] Connected devices (1 available)
    • iPhone • 662d19f06a36af09a9bd0e64f7b0fdcd83fc5167 • ios • iOS 11.1.2

! Doctor found issues in 2 categories.
```

Do you have any ideas what i can do to build flutter apps for ios11?

**Comment by cbracken** [comment]
@christian-muertz unfortunately I think the answer is that you’ll need to build with Xcode 9 to use the iOS 11 SDK, which supports safe area insets.
